src/commonMain/kotlin/com/ucasoft/kcron/builders/Builder.kt

```diff
@@ -82,29 +82,56 @@ class Builder(firstDayOfWeek: WeekDays = WeekDays.Monday) {
         return this
     }
 
-    fun nextRunList(maxCount: Int = 10) : List<LocalDateTime> {
-        val now = Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault())
-        val result = mutableListOf<LocalDateTime>()
-        for (year in (partBuilders.getValue(CronPart.Years) as YearsBuilder).years.filter { y -> y >= now.year }) {
-            for (month in (partBuilders.getValue(CronPart.Months) as MonthsBuilder).months.filter { m -> m >= now.monthNumber || year > now.year}) {
-                for (day in calculateDays(year, month, (partBuilders.getValue(CronPart.DaysOfWeek) as DaysOfWeekBuilder).daysOfWeek, (partBuilders.getValue(CronPart.Days) as DaysBuilder).days, now)) {
-                    for (hour in (partBuilders.getValue(CronPart.Hours) as HoursBuilder).hours.filter { h -> h >= now.hour || day > now.dayOfMonth || month > now.monthNumber || year > now.year }) {
-                        for (minute in (partBuilders.getValue(CronPart.Minutes) as MinutesBuilder).minutes.filter { m -> m >= now.minute || hour > now.hour || day > now.dayOfMonth || month > now.monthNumber || year > now.year }) {
-                            for (seconds in (partBuilders.getValue(CronPart.Seconds) as SecondsBuilder).seconds.filter { s -> s > now.second || minute > now.minute || hour > now.hour || day > now.dayOfMonth || month > now.monthNumber || year > now.year }) {
-                                result.add(LocalDateTime(year, month, day, hour, minute, seconds))
-                                if (result.size == maxCount) {
-                                    return result
+    @DelicateIterableApi
+    fun asIterable(start: LocalDateTime): Iterable<LocalDateTime> {
```

Comment:
Great improvement!
However, looks like we have to specify `start` default value as `now()`.

Reply:
> However, looks like we have to specify `start` default value as `now()`.

Implemented with method overloading in [f78076e2](https://github.com/Scogun/kcron-common/pull/20/commits/f78076e2c5962c2a6755893c97eb357afa04b524)